Server/src/controllers/dataUploadController.ts

```diff
@@ -53,9 +54,10 @@ export class DataUploadController {
         else {
             try {
                 this.dataSet = request.body
-                this.dataService = new EditUploadService(this.dataSet, datasetId, null)
-                await this.dataService.validateExtractedData();
-                let extractDataResponse: any = await this.dataService.uploadData();
+                let userId: number = request.body.user.account_id
+                let permissionLevel: number = request.body.user.account_admin
+                let uploadVerificationService = new EditUploadVerificationService()
+                let extractDataResponse: any = await uploadVerificationService.verifyUploader(userId, permissionLevel, this.dataSet, datasetId);
```

Comment:
I think it might be better to call the service you need ie editUploadServiice from the controller instead. This would respect the architecture we have setup. In general we should allow the controllers to call ALL the services he needs to complete the user request.